### DISCUSSION TITLE
Command not found

### DISCUSSION THREAD

**Discussion by NomarCub** [Taitava/obsidian-shellcommands#189] [Debugging and testing]
I have a fresh install of Linux (Pop!_OS 21.10) and specific commands are not found when executed in Obsidian, but work fine in external shells. I tried to set `code .` as a command this error pops up when I execute it: `[127]: /bin/sh: line 1: code: command not found`. The command works properly in the default OS terminal, both in sh and bash, the command makes a new VSCode window.

**Comment by Taitava** [comment]
Hi and thank you for your question!

Can you try to run the following command first in the normal terminal and then using the SC plugin (you can set the output channel to be e.g. [clipboard](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Clipboard)): `echo $PATH`.

In a nutshell, `$PATH` contains directories that contain all the executable programs that a shell knows. AFAIK, VSCode's directory should be there, too.

If you want, you can copy the outputs into this discussion. I understand it might be private information (contains file paths), so you don't have to do it if you don't want.

If the outputs differ from each other, then it probably means that VSCode's folder is not present in `$PATH` when SC is being used. We can then try to figure out why it's not there when SC is used.

Or there might be some other reason. Let's find out! 🙂

**Comment by Taitava** [comment]
# Related discussions
- A feature request related to `$PATH`: https://github.com/Taitava/obsidian-shellcommands/discussions/150
- A new discussion related to a similar problem: https://github.com/Taitava/obsidian-shellcommands/discussions/191

~~These discussions do not (at least yet) contain a solution to this problem, but I linked them just for reference.~~ Edit 2022-05-22: #150 is now in beta test.

**Reply by NomarCub** [reply]
Thank you! 
Unfortunately my fresh Linux install didn't last long, so I can't investigate this problem further now. I've saved this discussion for later, and will follow up on here when I have further issues or solutions.

**Reply by Taitava** [reply]
👍 You're welcome to come back anytime. 🙂

**Reply by Taitava** [reply]
Just a heads up here too, #150 is now in a beta test: https://github.com/Taitava/obsidian-shellcommands/discussions/228
